csharp/GSDK_CSharp_Standard/InternalSdk.cs

```diff
@@ -83,42 +85,31 @@ public Task StartAsync(bool debugLogs = false)
             Logger.Start();
             
 
-            string gsmsBaseUrl = ConfigMap[GameserverSDK.HeartbeatEndpointKey];
-            string instanceId = ConfigMap[GameserverSDK.ServerIdKey];
+            string heartbeatEndpoint = ConfigMap[GameserverSDK.HeartbeatEndpointKey];
+            string serverId = ConfigMap[GameserverSDK.ServerIdKey];
 
-            Logger.Log($"VM Agent Endpoint: {gsmsBaseUrl}");
-            Logger.Log($"Instance Id: {instanceId}");
+            Logger.Log($"VM Agent Endpoint: {heartbeatEndpoint}");
+            Logger.Log($"Instance Id: {serverId}");
 
-            _webClient = HttpClientFactory.CreateInstance($"http://{gsmsBaseUrl}/v1/sessionHosts/{instanceId}");
+            _httpClient = _httpClientFactory.CreateInstance($"http://{heartbeatEndpoint}/v1/sessionHosts/{serverId}");
 
-            _heartbeatRunning = true;
             _signalHeartbeatEvent.Reset();
             TransitionToActiveEvent.Reset();
 
             _heartbeatTask = Task.Run(HeartbeatAsync);
```

Comment:
>Task.Run(HeartbeatAsync); [](start = 29, length = 25)

Is this intended as fireAndForget?

Reply:
Yeah, that's my understanding, it just keeps running in the background. We should probably make it better by handling exceptions like we do on the VMAgent (RunUntilCancelled), but this refactoring seemed an improvement nonetheless.